### PR TITLE
switching buttons

### PR DESCRIPTION
**Pull request by mayabrandi**
This PR swhitces place on the buttns "Next page" and "First page" as described in this [issue](https://github.com/Clinical-Genomics/scout/issues/1367)
It also adds the functionality of next page to the cancer view

**How to test**:
1. Install on clinical db by running:
`bash servers/resources/clinical-db.scilifelab.se/update-scout-stage.sh new_switch_buttons`
2. Go to : https://scout-stage.scilifelab.se 
3. Look at the different variant views. The Next page button should be last. and the cancer view should have the new buttons at the bottom of the page.

**Review:**
- [x] code approved by dNil
- [x] tests executed by dNil

### CODE REVIEW

File: scout/server/blueprints/variants/views.py [hunk 1]
```diff
@@ -463,16 +463,17 @@ def clinvar(institute_id, case_name, variant_id):
...
     """Show cancer variants overview."""
     form = CancerFiltersForm(request.args)
-    data = controllers.cancer_variants(store, request.args, institute_id, case_name, form)
+    page = int(request.form.get('page', 1))
+    data = controllers.cancer_variants(store, request.args, institute_id, case_name, form, page=page)
     return data
 
 
-@variants_bp.route('/<institute_id>/<case_name>/<variant_id>/acmg', methods=['GET', 'POST'])
+@variants_bp.route('/<institute_id>/<case_name>/<variant_id>/acmg', methods=['POST'])
```
Comment (dnil) [inline scout/server/blueprints/variants/views.py]: Huh? No 'GET' for that one? There is an if clause for that in the function?

File: scout/server/blueprints/variants/controllers.py [hunk 2]
```diff
@@ -1097,17 +1097,25 @@ def verification_email_body(case_name, url, display_name, category, subcategory,
...
-def cancer_variants(store, request_args, institute_id, case_name, form):
+def cancer_variants(store, request_args, institute_id, case_name, form, page=1):
     """Fetch data related to cancer variants for a case."""
+
     institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
-    variants_query = store.variants(case_obj['_id'], category='cancer', query=form.data).limit(50)
+    per_page = 50
+    skip_count = per_page * max(page - 1, 0)
+    variants_query = store.variants(case_obj['_id'], category='cancer', query=form.data)
+    variant_count = variants_query.count()
```
Comment (dnil) [inline scout/server/blueprints/variants/controllers.py]: @moonso won't be happy about that one; ´´´.count()´´´ is apparently going deprecated in pymongo. But, we have the same for the other variants-controllers so we may as well change them all at once! Do keep it.

File: scout/server/blueprints/variants/templates/variants/str-variants.html [hunk 3]
```diff
@@ -131,22 +131,26 @@
...
+                </button>
+              </div>
           </div>
         </div>
       {% else %}
         <i class="text-muted">No more variants to display</i>
-        <a class="btn btn-outline-secondary mx-auto d-block" href="{{ url_for('variants.str_variants', institute_id=institute._id, case_name=case.display_name, page=1, **form.data) }}">
-          First page
-        </a>
+        <button class="btn btn-outline-secondary mx-auto d-block"
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/str-variants.html]: Button with href! Is that's new? :blush: I thought you had to either use an anchor, or use like an on_click function.

File: scout/server/blueprints/variants/templates/variants/cancer-variants.html [hunk 4]
```diff
@@ -88,9 +88,33 @@
...
     </table>
   </div><!-- end of card-->
+  <br>
+  <div class="container-fluid">
+      {{ pagination() }}
+  </div>
 </div><!--end of container-float -->
 {% endblock %}
 
+{% macro pagination() %}
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/cancer-variants.html]: :+1:

File: scout/server/blueprints/variants/templates/variants/str-variants.html (hunk 3 above)
Comment (dnil) [reply scout/server/blueprints/variants/templates/variants/str-variants.html]: Yeah, that doesn't work on stage. Either keep the link, adjust the form href or script the button. I would have done the first of the three, but anything that works! 😸

File: scout/server/blueprints/variants/views.py (hunk 1 above)
Comment (mayabrandi) [reply scout/server/blueprints/variants/views.py]: :) That was not intended at all. Ill take it back.

File: scout/server/blueprints/variants/controllers.py (hunk 2 above)
Comment (mayabrandi) [reply scout/server/blueprints/variants/controllers.py]: I see. What do we use instead?

File: scout/server/blueprints/variants/templates/variants/str-variants.html (hunk 3 above)
Comment (mayabrandi) [reply scout/server/blueprints/variants/templates/variants/str-variants.html]: 👍

File: scout/server/blueprints/variants/controllers.py (hunk 2 above)
Comment (mayabrandi) [reply scout/server/blueprints/variants/controllers.py]: Okej. Good to know

**Comment by mayabrandi** [comment]
All good now @dnil ?

**Comment by dnil** [comment]
I'm sure! Is it on stage already?

**Comment by dnil** [comment]
Sorry @mayabrandi; there is sth wrong with the cancer variants page, and I'm not so sure its your fault. The next/previous buttons do not work on the page. This appears to be an issue with filter argument passing - gene panel in particular. Do have a look, but it is not necessarily your code that is off. Feel free to edit the offending sections.. 😄

**Comment by dnil** [comment]
Otw all fine; STRs, SVs and regular SNVs test ok.

**Comment by mayabrandi** [comment]
Oh. Ill have a look right now. Strange though. Im sure i tested it both locally and on stage.

**Comment by dnil** [comment]
Thank you @mayabrandi, and sorry to have you start on something like that. Panels, and basically all POST requests handling was missing in the cancer variants view. Give it a spin now and see if isn't working better! When you feel ready to merge, please update the CHANGELOG!

**Comment by dnil** [comment]
On second thought: this now works as far as I can tell (tested on stage on SNV, SV, STR and cancer). I'll merge before there are any conflicts!

**Comment by northwestwitch** [comment]
Scared eh??? :D

**Comment by dnil** [comment]
> Scared eh??? :D

Winter is coming!

**Comment by mayabrandi** [comment]
Sorry for being silent for a couple of days. I did find the issue on the cancer view on friday, but dint manage to solve it then. But its all fixed now? Then ill leave it.

**Pull request merged**